packages/components/forms/src/controlled-input-field/ControlledInputField.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { EventHandler, ChangeEvent, ReactNode } from 'react';
+import { cx } from 'emotion';
+import { ValidationMessage } from '@contentful/f36-validation-message';
+import { Label, LabelProps } from '../Label';
+import { ControlledInput } from '@contentful/f36-inputs';
+import { HelpText } from '@contentful/f36-helptext';
+import type { ControlledInputProps } from '@contentful/f36-inputs';
+import { styles } from './ControlledInputField.styles';
+import { Box, BoxProps } from '@contentful/f36-core';
+
+export interface ControlledInputFieldProps extends BoxProps<'div'> {
+  id: string;
+  label: string;
+  isLabelLight?: boolean;
+  isRequired?: boolean;
+  helpText?: string;
+  formLabelProps?: LabelProps;
+  isDisabled?: boolean;
+  helpTextProps?: object;
+  validationMessage?: string;
+  value?: string;
+  name?: string;
+  isChecked?: boolean;
+  inputProps?: Omit<
+    ControlledInputProps,
+    | 'id'
+    | 'label'
+    | 'name'
+    | 'isRequired'
+    | 'isChecked'
+    | 'isDisabled'
+    | 'value'
+    | 'onChange'
+    | 'type'
+  >;
+  inputType?: 'radio' | 'checkbox';
+  onChange?: EventHandler<ChangeEvent<HTMLInputElement>>;
+  className?: string;
+  testId?: string;
+  children?: ReactNode;
+  ref: React.Ref<HTMLDivElement>;
```

Comment:
```suggestion
```

Is this used?

Reply:
Good point, the internal props don't contain a `ref` attribute. They can't since it will be captured by `forwardRef` and excluded from the `props` object. So for external usage, it's correct that we handle a `ref` prop while it's internally not really part of `props`. Let's just remove the `ref` definition from this type. I saw we currently have multiple approaches to export a PropsType, so we probably refactor all of them at some point at once.

Reply:
Ah, now I remember that removing this causes a typescript error. There are different definitions of ReactRef, a legacy one, and a more recent one and they are not compatible...
```
    // Used by forwardRef
    interface RefAttributes<T> extends Attributes {
        ref?: Ref<T>;
    }
    // Used by BoxProps
    interface ClassAttributes<T> extends Attributes {
        ref?: LegacyRef<T>;
    }
```